docs/faq.rst

```diff
@@ -65,6 +65,133 @@ Additional reading:
+You could see this if you use a ``print`` statement::
+
+  def func(x):
+    print(x)
+    return np.cos(x)
+
+  res = jax.jit(func)(0.)
+
+The above code does return the correct value ``1.`` but it also prints
+``Traced<ShapedArray(float32[])>`` for the value of ``x``. Normally, JAX
+handles these tracer values internally in a transparent way, e.g.,
+in the numeric JAX primitives that are used to implement the
+``jax.numpy`` functions. This is why ``np.cos`` works in the example above.
+
+More precisely, a **tracer** value is introduced for the argument of
+a JAX-transformed function, except the arguments identified by special
+parameters such as ``static_argnums`` for :func:`jax.jit` or
+``static_broadcasted_argnums`` for :func:`jax.pmap`. Typically, computations
+that involve at least a tracer value will produce a tracer value. Besides tracer
+values, there are **regular** Python values: values that are computed outside JAX
+transformations, or arise from above-mentioned static arguments of certain JAX
+transformations, or computed solely from other regular Python values.
+These are the values that are used everywhere in absence of JAX transformations.
+
+A tracer value carries an **abstract** value, e.g., ``ShapedArray`` with information
+about the shape and dtype of an array. Some tracers, e.g., those that are
+introduced for arguments of autodiff transformations, carry ``ConcreteArray``
+abstract values that actually include the regular array data, and are used,
+e.g., for resolving conditionals. Tracer values computed from these
+concrete tracers, perhaps in combination with regular values, result
+in **concrete tracers**. A **concrete value** is either a regular value or
+a concrete tracer.
+
+Most often values computed from tracer values are themselves tracer values.
+There are very few exceptions, when a computation can be entirely done
+using the abstract value carried by a tracer, in which case the result
+can be a regular value. For example, getting the shape of a tracer
+with ``ShapedArray`` abstract value. Another example, is when explicitly
+casting a concrete tracer value to a regular type: ``int(x)`` or ``float(x)``.
+Another such situation is for ``bool(x)``, which produces a Python bool when
+concreteness makes it possible. That case is especially salient because
+of how often it arises in control flow.
+
+Here is how the transformations introduce abstract or concrete tracers:
+
+  * :func:`jax.jit`: introduces **abstract tracers** for all positional arguments
+    except those denoted by ``static_argnums``, which remain regular
+    values.
+  * :func:`jax.pmap`: introduces **abstract tracers** for all positional arguments
+    except those denoted by ``static_broadcasted_argnums``.
+  * :func:`jax.vmap`, :func:`jax.make_jaxpr`, :func:`xla_computation`:
+    introduce **abstract tracers** for all positional arguments.
+  * :func:`jax.jvp` and :func:`jax.grad` introduce **concrete tracers**
```

Comment:
This isn't quite true, because `grad` might be called inside a `jit`, in which case the trace happens at shape level. It's more accurate to talk about levels of abstraction and say that transformations like `jit` and `vmap` raise the abstraction level to "shaped" while `jvp` doesn't raise it.

Reply:
I have added additional text to the explanation for `grad`.

Reply:
I agree with James that it's better to say "raise the level of abstraction". For example, instead of "introduces abstract tracers" how about "introduces tracers with abstract values raised to at least the ____ level", where ____ is different for different transformations?